regulations/static/components/Collapsible.vue

```diff
@@ -40,54 +44,61 @@ export default {
             required: false,
             default: "1s",
         },
-        direction: { //horizontal or vertical
+        direction: {
+            //horizontal or vertical
             type: String,
             required: true,
         },
     },
 
-    data: function() {
+    data: function () {
         return {
             size: 0,
             visible: true,
             isVertical: true,
             styles: {
                 overflow: "hidden",
                 transition: this.transition,
-            }
-        }
+            },
+        };
     },
 
     computed: {
-        sizeStyle: function() {
-            return this.isVertical ? 
-                { height: this.visible ? this.size : 0 } :
-                { width: this.visible ? this.size : 0 };
-        }
+        sizeStyle: function () {
+            return this.isVertical
+                ? { height: this.visible ? this.size : 0 }
+                : { width: this.visible ? this.size : 0 };
+        },
     },
 
     methods: {
-        resize: function(e) {
+        resize: function (e) {
             this.computeSize();
         },
-        toggle: function(target) {
-            if(this.name === target) {
-                this.visible = !this.visible;
+        toggle: function (target) {
+            if (this.name === target) {
+                if (!this.visible) {
+                    this.computeSize();
+                }
+                requestAnimationFrame(() => {
+                    this.visible = !this.visible;
```

Comment:
Moving the call to `computeSize` into this function might prevent add collision edge cases.

Reply:
This was added so that the transition will work when computing the size on the fly here when toggle is called.  I think the use of `requestAnimationFrame` in this context is similar to using `setTimeout` with a delay of `0` - it makes sure flipping the `visible` bool will not happen until everything in the event loop is done.  So `computeSize` needs to be outside the function because we need to make sure that's all settled and done before flipping `visible`, which triggers the transition animation.

Reply:
Could there be an outside edge case where the size is computed, a change is made that would change the result, and then `visible` is flipped?